### DISCUSSION TITLE
How to use Oracle.ManagedDataAccess.Core driver with Nhibernate 5 targeting .NET 5?

### DISCUSSION THREAD

**Discussion by maulik-modi** [nhibernate/nhibernate-core#2907] [Q&A] [answered]
How can we use https://www.nuget.org/packages/Oracle.ManagedDataAccess.Core/ for project targeting .NET 5(Cross platform deployment) and Oracle 12c Database?

I can see that https://github.com/nhibernate/nhibernate-core/tree/master/src/NHibernate/Driver folder has OracleManagedDataClientDriver.cs file using Oracle.ManagedDataAccess driver. 
```
namespace NHibernate.Driver
{
	/// <summary>
	/// A NHibernate Driver for using the Oracle.ManagedDataAccess DataProvider
	/// </summary>
	public class OracleManagedDataClientDriver : OracleDataClientDriverBase
	{
		/// <summary>
		/// Initializes a new instance of <see cref="OracleManagedDataClientDriver"/>.
		/// </summary>
		/// <exception cref="HibernateException">
		/// Thrown when the <c>Oracle.ManagedDataAccess</c> assembly can not be loaded.
		/// </exception>
		public OracleManagedDataClientDriver()
			: base("Oracle.ManagedDataAccess")
		{
		}

		public override bool HasDelayedDistributedTransactionCompletion => true;
	}
}
```

Questions

1. Do I simply create a new class inheriting from OracleDataClientDriverBase and referencing "Oracle.ManagedDataAccess.Core" and it works OR there are some other changes required to get it working? 
2. How about Nhibernate 5 comatibility with this new driver - Oracle.ManagedDataAccess.Core , I can see that @bahusoid  PR https://github.com/nhibernate/nhibernate-core/pull/2848 has been merged, Does it mean most tests are being passed in here? What dialect https://nhibernate.info/doc/nhibernate-reference/session-configuration.html#configuration-optional-dialects are you using in your tests?
![image](https://user-images.githubusercontent.com/34910430/132189438-c06e3350-e953-4618-ad67-ad51a4ebaa83.png)

@bahusoid , @fredericDelaporte , @hazzik , your thoughts?

**Comment by hazzik** [comment, marked as answer]
Just set "connection.driver_class" to  "NHibernate.Driver.OracleManagedDataClientDriver" and "dialect" to "NHibernate.Dialect.Oracle12cDialect" in your configuration file. Don't need to create any classes or anything.

```xml
<hibernate-configuration  xmlns="urn:nhibernate-configuration-2.2" >
	<session-factory name="NHibernate.Test">
		<property name="connection.driver_class">NHibernate.Driver.OracleManagedDataClientDriver</property>
		<property name="connection.connection_string">
			User ID=nhibernate;Password=nhibernate;Data Source=(DESCRIPTION = (ADDRESS = (PROTOCOL = TCP)(HOST = localhost)(PORT = 1521)) (CONNECT_DATA = (SERVER = DEDICATED) (SERVICE_NAME = ORCL)))
		</property>
		<property name="show_sql">false</property>
		<property name="dialect">NHibernate.Dialect.Oracle12cDialect</property>
		<property name="query.substitutions">true 1, false 0, yes 'Y', no 'N'</property>
		<!-- If your database setup use an ASCII charset, switch following property to true. -->
		<property name="oracle.use_n_prefixed_types_for_unicode">false</property>
		<!-- Depending on your database setup, the default cast length of 4000 may be too big.
		     By example, if previous setting is true, NHibernate may try to use nvarchar2(4000),
		     which will be rejected if its underlying charset is UTF16 and the database
		     MAX_STRING_SIZE is not extended. In such case, reduce it to 2000. -->
		<property name="query.default_cast_length"></property>
	</session-factory>
</hibernate-configuration>

```

We run 2.19.x driver just because our tests still target .NET Core 2.0 as we've been lazy to upgrade the test suite. I've run locally .NET 6.0 and most test do work as expected.
